src/xpk/core/storage.py

```diff
@@ -257,6 +258,131 @@ def install_storage_crd(k8s_api_client: ApiClient) -> None:
       xpk_exit(1)
 
 
+def get_storage_volume_mounts_yaml(storages: list[Storage]) -> str:
+  """
+  Generates the YAML representation of the volumeMounts section for the given Storages.
+
+  This function creates the YAML snippet that defines how the storage volumes
+  should be mounted within a Pod's containers.
+
+  Args:
+      storages: A list of Storage objects.
+
+  Returns:
+      A string containing the YAML representation of the volumeMounts section.
+  """
+  yaml_str = ""
+  for storage in storages:
+    yaml_str += f"""- name: {storage.pv}
+                mountPath: {storage.mount_point}
+                readOnly: {storage.readonly}
+            """
+  return yaml_str
+
+
+def get_storage_volumes_yaml(storages: list[Storage]) -> str:
+  """
+  Generates the YAML representation of the volumes section for the given Storages.
+
+  This function creates the YAML snippet that defines the volumes to be
+  mounted in a Pod, including the PersistentVolumeClaim associated with
+  each Storage.
+
+  Args:
+      storages: A list of Storage objects.
+
+  Returns:
+      A string containing the YAML representation of the volumes section.
+  """
+  yaml_str = ""
+  for storage in storages:
+    yaml_str += f"""- name: {storage.pv}
+              persistentVolumeClaim:
+                claimName: {storage.pvc}
+                readOnly: {storage.readonly}
+            """
+  return yaml_str
+
+
+def get_storage_volume_mounts_yaml_for_gpu(storages: list[Storage]) -> str:
+  """
+  Generates the YAML representation of the volumeMounts section for the given Storages.
+
+  This function creates the YAML snippet that defines how the storage volumes
+  should be mounted within a Pod's containers.
+
+  Args:
+      storages: A list of Storage objects.
+
+  Returns:
+      A string containing the YAML representation of the volumeMounts section.
+  """
+  yaml_str = ""
+  for storage in storages:
+    yaml_str += f"""- name: {storage.pv}
+                  mountPath: {storage.mount_point}
+                  readOnly: {storage.readonly}
+            """
+  return yaml_str
+
+
+def get_storage_volumes_yaml_for_gpu(storages: list[Storage]) -> str:
+  """
+  Generates the YAML representation of the volumes section for the given Storages.
+
+  This function creates the YAML snippet that defines the volumes to be
+  mounted in a Pod, including the PersistentVolumeClaim associated with
+  each Storage.
+
+  Args:
+      storages: A list of Storage objects.
+
+  Returns:
+      A string containing the YAML representation of the volumes section.
+  """
+  yaml_str = ""
+  for storage in storages:
```

Comment:
if there are multiple storages shouldn't we append \n on end of each persistentVolumeClaim?